Add per-tenant max cache freshness to query-frontend (#2609)
cortexproject/cortex: CHANGELOG.md

* [CHANGE] Query Frontend now uses Round Robin to choose a tenant queue to service next. #2553
* [FEATURE] TLS config options added for GRPC clients in Querier (Query-frontend client & Ingester client), Ruler, Store Gateway, as well as HTTP client in Config store client. #2502

> * [FEATURE] The flag `frontend.max-cache-freshness` can now be used to specify per-tenant max cache freshness values. The corresponding YAML config parameter has been changed from `results_cache.max_freshness` to `limits_config.max_cache_freshness`. The legacy YAML config parameter (`results_cache.max_freshness`) will continue to be supported till Cortex release `v1.4.0`. #2609

Reviewer: Please always prefix flags with `-` for clarity. I've also reworded it a bit, because overrides are possible only via YAML.

```suggestion
* [FEATURE] The flag `-frontend.max-cache-freshness` is now supported within the limits overrides, to specify per-tenant max cache freshness values. The corresponding YAML config parameter has been changed from `results_cache.max_freshness` to `limits_config.max_cache_freshness`. The legacy YAML config parameter (`results_cache.max_freshness`) will continue to be supported till Cortex release `v1.4.0`. #2609
```
